tests/foreman/cli/test_host.py

```diff
@@ -25,66 +37,497 @@ def setUp(self):
         self.puppet_proxy = result[0]
 
     @tier1
-    def test_positive_create_1(self):
+    def test_positive_create_with_name(self):
         """@test: A host can be created with a random name
 
         @feature: Hosts
 
         @assert: A host is created and the name matches
+        """
+        for name in valid_hosts_list():
+            with self.subTest(name):
+                host = entities.Host()
+                host.create_missing()
+                result = Host.create({
+                    u'architecture-id': host.architecture.id,
+                    u'domain-id': host.domain.id,
+                    u'environment-id': host.environment.id,
+                    # pylint:disable=no-member
+                    u'location-id': host.location.id,
+                    u'mac': host.mac,
+                    u'medium-id': host.medium.id,
+                    u'name': name,
+                    u'operatingsystem-id': host.operatingsystem.id,
+                    # pylint:disable=no-member
+                    u'organization-id': host.organization.id,
+                    u'partition-table-id': host.ptable.id,
+                    u'puppet-proxy-id': self.puppet_proxy['id'],
+                    u'root-pass': host.root_pass,
+                })
+                self.assertEqual(
+                    '{0}.{1}'.format(name, host.domain.read().name).lower(),
+                    result['name'],
+                )
+
+    @tier2
```

Comment:
It seems more as tear1 as you don't perform association, but actually create entity (no matter what parameters you use)